src/Simplify.elm

```diff
@@ -6814,24 +6754,144 @@ collectionMapChecks collection checkInfo =
 
                 Nothing ->
                     []
-        , \() ->
-            if AstHelpers.isIdentity checkInfo.lookupTable checkInfo.firstArg then
-                [ Rule.errorWithFix
-                    { message = "Using " ++ collection.moduleName ++ ".map with an identity function is the same as not using " ++ collection.moduleName ++ ".map"
-                    , details = [ "You can remove this call and replace it by the " ++ collection.represents ++ " itself." ]
-                    }
-                    checkInfo.fnRange
-                    (toIdentityFix
-                        { lastArg = maybeListArg, resources = checkInfo }
-                    )
-                ]
-
-            else
-                []
         ]
         ()
 
 
+{-| TODO merge with identityError
+-}
+mapIdentityChecks :
+    { a
```

Comment:
We don't an extensible record here.

Reply:
It's true that we currently don't need it but if you look at for example
```elm
collectionMapChecks collection checkInfo =
    ...
            mapIdentityChecks
                { moduleName = AstHelpers.moduleNameFromString collection.moduleName
                , represents = collection.represents
                }
                checkInfo
```
you can see that the extensibility only accidentally doesn't help because for some reason `Collection` takes `moduleName` as a `String`. How about I make the two compatible in the next cleanup PR?

Reply:
Sounds good :+1: